### PR TITLE
[WFCORE-6466] Split LayersTest#test into two separate methods

### PR DESCRIPTION
**Pull request by The-Huginn**
JIRA: https://issues.redhat.com/browse/WFCORE-6466

### CODE REVIEW

File: testsuite/shared/src/main/java/org/jboss/as/test/layers/LayersTest.java [hunk 1]
```diff
@@ -71,31 +71,68 @@ public class LayersTest {
      * @param unused The set of modules that are OK to not be provisioned when all layers are provisioned.
      * If more modules than this set are not provisioned, it means that we are missing some modules and
      * an error occurs.
-     * @throws Exception
+     * @throws Exception on failure
      */
+    @Deprecated
     public static void test(String root, Set<String> unreferenced, Set<String> unused) throws Exception {
+        testBootInstallations(root);
```
Comment (bstansberry) [inline testsuite/shared/src/main/java/org/jboss/as/test/layers/LayersTest.java:78]: This should be a call to the existing testExecutions(String) as the new testBootInstallations does the same thing.

File: testsuite/shared/src/main/java/org/jboss/as/test/layers/LayersTest.java [hunk 2]
```diff
@@ -71,31 +71,68 @@ public class LayersTest {
...
-                    layers.put(f.getName(), res);
                 }
             }
         } finally {
             executor.shutdownNow();
         }
+    }
+
+    /**
+     * Checks only allowed modules were loaded with the installation
```
Comment (bstansberry) [inline testsuite/shared/src/main/java/org/jboss/as/test/layers/LayersTest.java:105]: I'd say

"Checks the expected modules were provisioned with the @{code test-standalone-reference} and @{test-all-layers} installations."

**Review by bstansberry — CHANGES_REQUESTED** [review]
Thanks for looking at this @The-Huginn. I think it can be simplified a bit, but this is what I had in mind.

File: testsuite/shared/src/main/java/org/jboss/as/test/layers/LayersTest.java (hunk 1 above)
Comment (The-Huginn) [reply testsuite/shared/src/main/java/org/jboss/as/test/layers/LayersTest.java:78]: Sorry, I didn't realize that

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @The-Huginn

**Pull request merged**